src/Microsoft.ML.Data/DataLoadSave/Text/TextLoaderSaverCatalog.cs

```diff
@@ -16,35 +20,126 @@ public static class TextLoaderSaverCatalog
+        public static TextLoader CreateTextReader(this DataOperations catalog,
+            TextLoader.Arguments args,
+            IMultiStreamSource dataSample = null)
             => new TextLoader(CatalogUtils.GetEnvironment(catalog), args, dataSample);
 
+        /// <summary>
+        /// Create a text reader <see cref="TextLoader"/> by inferencing the dataset schema from a data model type.
+        /// </summary>
+        /// <param name="catalog">The <see cref="DataOperations"/> catalog.</param>
+        /// <param name="hasHeader">Does the file contains header?</param>
+        /// <param name="separator">Column separator character. Default is '\t'</param>
+        /// <param name="allowQuotedStrings">Whether the input may include quoted values,
+        /// which can contain separator characters, colons,
+        /// and distinguish empty values from missing values. When true, consecutive separators
+        /// denote a missing value and an empty value is denoted by \"\".
+        /// When false, consecutive separators denote an empty value.</param>
+        /// <param name="supportSparse">Whether the input may include sparse representations for example,
+        /// if one of the row contains "5 2:6 4:3" that's mean there are 5 columns all zero
+        /// except for 3rd and 5th columns which have values 6 and 3</param>
+        /// <param name="trimWhitespace">Remove trailing whitespace from lines</param>
+        public static TextLoader CreateTextReader<TInput>(this DataOperations catalog,
+            bool hasHeader = TextLoader.DefaultArguments.HasHeader,
+            char separator = TextLoader.DefaultArguments.Separator,
+            bool allowQuotedStrings = TextLoader.DefaultArguments.AllowQuoting,
+            bool supportSparse = TextLoader.DefaultArguments.AllowSparse,
+            bool trimWhitespace = TextLoader.DefaultArguments.TrimWhitespace)
+        {
+            var userType = typeof(TInput);
+
+            var fieldInfos = userType.GetFields(BindingFlags.Public | BindingFlags.Instance);
+
+            var propertyInfos =
+                userType
+                .GetProperties(BindingFlags.Public | BindingFlags.Instance)
+                .Where(x => x.CanRead && x.CanWrite && x.GetGetMethod() != null && x.GetSetMethod() != null && x.GetIndexParameters().Length == 0);
+
+            var memberInfos = (fieldInfos as IEnumerable<MemberInfo>).Concat(propertyInfos).ToArray();
+
+            var columns = new TextLoader.Column[memberInfos.Length];
+
+            for (int index = 0; index < memberInfos.Length; index++)
+            {
+                var memberInfo = memberInfos[index];
+                var mappingAttr = memberInfo.GetCustomAttribute<LoadColumnAttribute>();
+                var mptr = memberInfo.GetCustomAttributes();
+
+                Contracts.Assert(mappingAttr != null, $"Field or property {memberInfo.Name} is missing the LoadColumn attribute");
```

Comment:
>LoadColumn [](start = 107, length = 10)

maybe `nameof` ?

Reply:
Using this comment to ask a meta quesiton: does it make more sense to continue that to throw? Maybe not all the properties of the data model will be decorated with LoadColumn. @Zruty0, @tomfinly

---
In reply to: [242670646](https://github.com/dotnet/machinelearning/pull/1878#discussion_r242670646) [](ancestors = 242670646)

Reply:
Resolving the questions, since the discussion has moved to the other comment. 

---
In reply to: [243040913](https://github.com/dotnet/machinelearning/pull/1878#discussion_r243040913) [](ancestors = 243040913,242670646)